src/material-experimental/mdc-slider/slider.ts

```diff
@@ -0,0 +1,517 @@
+    'class': 'mat-mdc-slider mdc-slider mdc-slider--display-markers',
+    'role': 'slider',
+    'aria-orientation': 'horizontal',
+    // The tabindex if the slider turns disabled is managed by the MDC foundation which
+    // dynamically updates and restores the "tabindex" attribute.
+    '[attr.tabindex]': 'tabIndex || 0',
+    '[class.mdc-slider--discrete]': 'thumbLabel',
+    '[class.mat-slider-has-ticks]': 'tickInterval !== 0',
+    '[class.mat-slider-thumb-label-showing]': 'thumbLabel',
+    '[class.mat-slider-disabled]': 'disabled',
+    '[class.mat-primary]': 'color == "primary"',
+    '[class.mat-accent]': 'color == "accent"',
+    '[class.mat-warn]': 'color == "warn"',
+    '[class._mat-animation-noopable]': '_animationMode === "NoopAnimations"',
+    '(blur)': '_markAsTouched()',
+  },
+  exportAs: 'matSlider',
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [MAT_SLIDER_VALUE_ACCESSOR],
+})
+export class MatSlider implements AfterViewInit, OnChanges, OnDestroy, ControlValueAccessor {
+  /** Event emitted when the slider value has changed. */
+  @Output() readonly change: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /** Event emitted when the slider thumb moves. */
+  @Output() readonly input: EventEmitter<MatSliderChange> = new EventEmitter<MatSliderChange>();
+
+  /**
+   * Emits when the raw value of the slider changes. This is here primarily
+   * to facilitate the two-way binding for the `value` input.
+   * @docs-private
+   */
+  @Output() readonly valueChange: EventEmitter<number> = new EventEmitter<number>();
+
+  /** Tabindex for the slider. */
+  @Input() tabIndex: number = 0;
+
+  /** The color palette for this slider. */
+  @Input() color: ThemePalette = 'accent';
+
+  /**
+   * Function that will be used to format the value before it is displayed
+   * in the thumb label. Can be used to format very large number in order
+   * for them to fit into the slider thumb.
+   */
+  @Input() displayWith: (value: number) => string | number;
+
+  /** The minimum value that the slider can have. */
+  @Input()
+  get min(): number {
+    return this._min;
+  }
+  set min(value: number) {
+    this._min = coerceNumberProperty(value);
+  }
+  private _min = 0;
+
+  /** The maximum value that the slider can have. */
+  @Input()
+  get max(): number {
+    return this._max;
+  }
+  set max(value: number) {
+    this._max = coerceNumberProperty(value);
+  }
+  private _max = 100;
+
+  /** Value of the slider. */
+  @Input()
+  get value(): number|null {
+    // If the value needs to be read and it is still uninitialized, initialize
+    // it to the current minimum value.
+    if (this._value === null) {
+      this.value = this.min;
+    }
+    return this._value;
+  }
+  set value(value: number|null) {
+    const coercedValue = coerceNumberProperty(value);
+    if (coercedValue !== this._value) {
```

Comment:
What purpose does this check serve?

Reply:
Nothing really other than just avoiding the assignment. It's more of less what is done in the standard Angular
Material slider right now. Will remove it.